Add release blogs page (#4836)
FlowFuse/website: src/handbook/engineering/releases/release-blogs.md

---
navTitle: Release Blogs
---

# Writing Release Blogs

> The [FlowFuse blog](/blog/) is where users, customers, and the broader Node-RED community go to understand what has changed in a release. A release blog is published on release day and covers the full scope of a release at a level of depth the changelog does not.

Reviewer: ```suggestion
The [FlowFuse blog](/blog/releases/) is where users, customers, and the broader Node-RED community go to understand what has changed in a release. A release blog is published on release day and covers the full scope of a release at a level of depth the changelog does not.
```

**2. Feature content**
Use H3 headings when multiple features are grouped under one H2. Each feature gets 1–2 factual paragraphs. No preamble. No "we're excited to announce."

**3. `### In practice`**
Three bullet points, each starting with "You." Describe user outcomes, not feature capabilities.

| Wrong (capability) | Right (outcome) |
|---|---|
| **Push and pull snapshots** directly from your repositories | Your Node-RED flows participate in the same version control workflow as the rest of your stack |
| **Use Azure Personal Access Tokens** for authentication | You authenticate with Azure Personal Access Tokens, with no secondary tooling required |

### `## What else is new?`

Bullet list of smaller improvements, fixes, and Node-RED updates. No "In practice" needed. One sentence per item. Group fixes under a `### Fixes` subheading.

### One CTA only

Keep the post to a single, well-defined call to action. Multiple CTAs split the reader's attention and reduce the effectiveness of each.

> The CTA is already handled via front matter — that is where it should be tailored if you need to customise it for a release. Do not add a `## Try FlowFuse` block manually in the body. It will render twice on the published page.

Reviewer: ```suggestion
The CTA is already handled via front matter. For guidance on how to add or customize it, refer to the handbook: [CTA Guidelines](handbook/marketing/blog/#cta). Do not add a `## Try FlowFuse` block manually in the body, as it will render twice on the published page.
```

> The CTA is already handled via front matter. For guidance on how to add or customize it, refer to the handbook: [CTA Guidelines](handbook/marketing/blog/#cta). Do not add a `## Try FlowFuse` block manually in the body, as it will render twice on the published page.

Reviewer: ```suggestion
The CTA is already handled via front matter. For guidance on how to add or customize it, refer to the handbook: [CTA Guidelines](/handbook/marketing/blog/#cta). Do not add a `## Try FlowFuse` block manually in the body, as it will render twice on the published page.
```

Reviewer: added missing `/`

```
src/blog/YYYY/MM/flowfuse-release-X-YY.md
```

If the folder for that month does not yet exist, create it.

## Frontmatter

Every post requires the following fields:

| Field | Notes |
|---|---|
| `title` | Format: "FlowFuse X.Y: [Short theme]". Title case. The subtitle after the colon should name the one or two most significant things in the release. |
| `subtitle` | One sentence expanding on the title. Names the next tier of features if the title does not cover them. |
| `description` | One sentence for link previews and search results. Should make sense without surrounding context. |
| `date` | Release date in `YYYY-MM-DD` format. |
| `authors` | Your handle from `src/_data/team`. |
| `image` | Path to the hero image. Coordinate with design. |
| `tags` | Always include `flowfuse`, `news`, and `releases`. |
| `release` | The release number as a string, e.g. `"2.29"`. |
| `features` | List of feature anchors for the in-page navigation. Each entry needs a `heading` and, where a changelog entry exists, an `id` matching the feature catalog slug. |

> Before populating `features`: Confirm that `featureCatalog.yml` has been updated for every feature in this release. The `id` values here must match slugs in the catalog — if a feature is missing from the catalog, the in-page navigation will silently break and tier badges will not render. The changelog posts for each feature should have already triggered this update; if any are missing, flag them before the blog goes live. See [Writing Changelog Posts](writing-changelog/#writing-changelog-posts) for how catalog entries work.

Reviewer: ```suggestion
Before populating `features`: Confirm that `featureCatalog.yml` has been updated for every feature in this release. The `id` values here must match slugs in the catalog — if a feature is missing from the catalog, the in-page navigation will silently break and tier badges will not render. The changelog posts for each feature should have already triggered this update; if any are missing, flag them before the blog goes live. See [Writing Changelog Posts](/handbook/engineering/releases/writing-changelog/#feature-catalog-and-availability) for how catalog entries work.
```

**Right:**
> FlowFuse 2.29 gives teams more control over how flows move through their stack, makes it easier to understand what changed between versions, and brings FlowFuse Expert to self-hosted enterprise customers.

### H2 sections

Group features under outcome-oriented H2 headings, not feature names or product labels. Two features that solve the same user problem belong in the same section. A release with three features might need two H2 sections, not three.

| Wrong | Right |
|---|---|
| `## Azure DevOps Git Integration` | `## More Visibility and Control Across Your Deployment Workflow` |
| `## FlowFuse Expert` | `## FlowFuse Expert, Available to More Teams and More Capable` |

Each H2 section contains:

**1. Problem framing (1–2 sentences)**
What was the friction or gap before this? Write it as a user experience.

> Managing flows across environments means tracking what changed, when, and by whom. When tooling gaps introduce friction here — or leave your version control workflow fragmented — they slow teams down at exactly the wrong moment.

**2. Feature content**

> Use H3 headings when multiple features are grouped under one H2. Each feature gets 1–2 factual paragraphs. No preamble. No "we're excited to announce."

Reviewer 2: @allthedoll for the H3 it would be expected to include the feature name, I think it would be worth mentioning it here.

Be specific. Vague sections are useless.

**Weak:** Improvements to the snapshot comparison view.

**Strong:** The compare dialog now includes a property-level diff sidebar: structural property changes old to new at a glance, and git-style line diffs for function code, template HTML, and JSON.

Keep paragraphs short. One idea per sentence. If a sentence contains more than one clause joined by "and," it probably needs to be two sentences.

Sentence case everywhere — headings, bullets, captions.

Do not bold feature names mid-sentence. Use plain prose or `code style` for technical terms.

Do not repeat in "In practice" what you already said in the prose above it.

Avoid marketing language. Do not use: exciting, powerful, seamless, robust, game-changing, best-in-class. If the feature is good, the outcome statement will demonstrate it.

## Drafting with AI

Give an AI tool your feature list — not a written draft. If you hand it a draft, it will polish rather than restructure. A bullet list of what shipped forces the grouping and framing work to happen from scratch.

> Paste the following prompt along with your feature list:

Reviewer 2: @allthedoll we now have the skill you created, is this full prompt still necessary?
